subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt

```diff
@@ -184,28 +179,18 @@ public abstract class InstrumentationTestsTask @Inject constructor(
                 applicationProguardMapping,
                 testProguardMapping
             ).mapNotNull { it.orNull?.asFile },
-            useInMemoryReport = useInMemoryReport.get(),
-            uploadTestArtifacts = uploadAllTestArtifacts.get(),
+            useInMemoryReport = experiments.useInMemoryReport,
+            uploadTestArtifacts = experiments.uploadArtifactsFromRunner,
             reportViewerConfig = reportViewerConfig,
-            fetchLogcatForIncompleteTests = fetchLogcatForIncompleteTests.get(),
-            saveTestArtifactsToOutputs = saveTestArtifactsToOutputs.get(),
+            fetchLogcatForIncompleteTests = experiments.fetchLogcatForIncompleteTests,
+            saveTestArtifactsToOutputs = experiments.saveTestArtifactsToOutputs,
         )
 
-        if (testRunnerService.isPresent) {
-            val service = testRunnerService.get()
-            val queue = workerExecutor.noIsolation()
-
-            queue.submit(TestRunnerWorkAction::class.java) { params ->
-                params.service.set(service)
-                params.statsDConfig.set(statsDConfig)
-                params.testRunParams.set(
-                    TestRunParams(
-                        projectName = project.name,
-                        instrumentationConfigName = configuration.name
-                    )
-                )
-                params.legacyTestRunParams.set(testRunParams)
-            }
+        if (dumpParams.getOrElse(false)) {
+            RunnerInputTester.dumpInput(
+                rootDir = project.rootDir,
+                input = testRunParams
+            )
```

Comment:
Proposal: make it always before any execution.
Like we do it in `FilterInfoWriter`.

It will help in troubleshooting. 
Anyone could check manually that the runner received expected parameters.
Without this information, it's harder to check that all configs in buildscripts are converted to expected input for the test runner.

Reply:
I like the idea, let's do it

Reply:
Current dump is a bin file (result of java serialization)
I have no plans to change it, because it is more safe than asserting on extra serialization mechanism (like to json)
What do you think? add json alongside with bin?

Reply:
From what I know now, yes, it's worth it.
But better in additional PR.